src/Symfony/Component/Clock/MockClock.php

```diff
@@ -80,19 +93,47 @@ public function modify(string $modifier): void
      */
     public function withTimeZone(\DateTimeZone|string $timezone): static
     {
+        $this->now ??= $this->tryFromTimeProvider();
+
         if (\PHP_VERSION_ID >= 80300 && \is_string($timezone)) {
-            $timezone = new \DateTimeZone($timezone);
+            $this->timezone = new \DateTimeZone($timezone);
         } elseif (\is_string($timezone)) {
             try {
-                $timezone = new \DateTimeZone($timezone);
+                $this->timezone = new \DateTimeZone($timezone);
             } catch (\Exception $e) {
                 throw new \DateInvalidTimeZoneException($e->getMessage(), $e->getCode(), $e);
             }
         }
 
         $clone = clone $this;
-        $clone->now = $clone->now->setTimezone($timezone);
+        $clone->timeProvider = $this->timeProvider;
+        $clone->timezone = $this->timezone;
+        $clone->now = $clone->now->setTimezone($this->timezone);
 
         return $clone;
     }
+
+    public function reset(): void
+    {
+        if (null === $this->timeProvider) {
+            throw new \LogicException('The MockClock can only be reset when it was created with a time provider closure.');
```

Comment:
I don't think this exception is useful, it's rather going to be annoying to me

Reply:
what should be the effect of resetting a MockClock created with a fixed date ?

Reply:
it should do nothing - no-op